api/v2beta1/ratelimitpolicy_types.go

```diff
@@ -0,0 +1,218 @@
+/*
+Copyright 2021 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v2beta1
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/go-logr/logr"
+	"github.com/google/go-cmp/cmp"
+	"github.com/kuadrant/kuadrant-operator/pkg/common"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	gatewayapiv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// ContextSelector defines one item from the well known selectors
+// TODO Document properly "Well-known selector" https://github.com/Kuadrant/architecture/blob/main/rfcs/0001-rlp-v2.md#well-known-selectors
+// +kubebuilder:validation:MinLength=1
+// +kubebuilder:validation:MaxLength=253
+type ContextSelector string
+
+// +kubebuilder:validation:Enum:=eq;neq
+type WhenConditionOperator string
+
+// +kubebuilder:validation:Enum:=second;minute;hour;day
+type TimeUnit string
+
+// Rate defines the actual rate limit that will be used when there is a match
+type Rate struct {
+	// Limit defines the max value allowed for a given period of time
+	Limit int `json:"limit"`
+
+	// Duration defines the time period for which the Limit specified above applies.
+	Duration int `json:"duration"`
+
+	// Duration defines the time uni
+	// Possible values are: "second", "minute", "hour", "day"
+	Unit TimeUnit `json:"unit"`
+}
+
+// RouteSelector defines semantics for matching an HTTP request based on conditions
+// https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPRouteSpec
+type WhenCondition struct {
+	// Selector defines one item from the well known selectors
+	// TODO Document properly "Well-known selector" https://github.com/Kuadrant/architecture/blob/main/rfcs/0001-rlp-v2.md#well-known-selectors
+	Selector ContextSelector `json:"selector"`
+
+	// The binary operator to be applied to the content fetched from the selector
+	// Possible values are: "eq" (equal to), "neq" (not equal to), "incl" (includes; for arrays), "excl" (excludes; for arrays), "matches" (regex)
+	Operator WhenConditionOperator `json:"operator"`
+
+	// The value of reference for the comparison.
+	// If used with the "matches" operator, the value must compile to a valid Golang regex.
+	Value string `json:"value"`
```

Comment:
Authorino conditions support only string comparisons too. [I've been wondering](https://github.com/Kuadrant/authorino/issues/357#issuecomment-1502904440) lately if we shouldn't try to support any JSON/YAML value type. In the future, this may be helpful for numeric operators (e.g. greater than, lower than, etc), array lookups, etc. WDYT?

Reply:
Good point. For now, as the condition operator is either `eq` or `neq`, I would stick to literal (string).